### DISCUSSION TITLE
Option to not generate the @WebService classes

### DISCUSSION THREAD

**Discussion by jarTaracz** [eclipse-ee4j/metro-jax-ws#346] [Tools]
Is there a way to configure the jaxws-maven-plugin to not generate the `@WebService` classes? When I use xjc tool it generates `@XmlType` classes only (and that is what I need, but xjc tool is not an option for me anymore). For example with wsdl file given [here](https://www.tutorialworks.com/wsdl/) I don't want the `BookService` interface and `BookService_Service` class to be generated. My configuration is:
```` 
<plugin>
    <groupId>com.sun.xml.ws</groupId>
    <artifactId>jaxws-maven-plugin</artifactId>
    <version>2.3.5</version>
    <executions>
        <execution>
            <id>generate-test</id>
            <goals>
                <goal>wsimport</goal>
            </goals>
            <configuration>
                <wsdlDirectory>${test.wsdl.dir}</wsdlDirectory>
                <wsdlFiles>
                    <wsdlFile>test.wsdl</wsdlFile>
                </wsdlFiles>
                <sourceDestDir>${project.basedir}/src/main/java</sourceDestDir>
                <packageName>${test.generated.package}</packageName>
                <encoding>utf-8</encoding>
                <xnocompile>true</xnocompile>
            </configuration>
        </execution>
    </executions>
</plugin>
````

**Comment by lukasj** [comment]
no; `xjc` is what you need; why `xjc` is no longer an option for you?

**Reply by jarTaracz** [reply]
Thank you for your response. The problem is that my team has agreed that the configuration of our project should be as simple as possible and should not require any additional tool installed. Also project should be prepared to migration to java 11. xjc is not included in JDK 11. Also we want to easily migrate from `javax.xml.*` to `jakarta.xml.*` packages. So `jaxws-maven-plugin` seems to be a perfect choice, I guess those additionally generated classes is the price we have to pay :)

**Reply by lukasj** [reply]
there are maven plugins for xjc as well - ~~see section 7.1.5 in the [User Guide](https://eclipse-ee4j.github.io/jaxb-ri/4.0.0/docs/ch03.html#deployment-maven-coordinates)~~

**Reply by lukasj** [reply]
section about jaxb-maven-plugins has been updated in the jaxb user guide, see [3rd party tools](https://eclipse-ee4j.github.io/jaxb-ri/4.0.0/docs/ch04.html#tools-3rd-party-tools) there

**Reply by jarTaracz** [reply]
I've decided to use [highsource maven-jaxb2-plugin](https://github.com/highsource/maven-jaxb2-plugin). Thank you for listing those plugins in the manual. I really appreciate your help.
